### PR TITLE
[Tapioca add-on] Add `ServerAddon` boilerplate

### PR DESCRIPTION
**Pull request by KaanOzkan**
### Motivation

Add ServerAddon usage to be called by Rails app based on https://github.com/Shopify/tapioca/compare/main...andyw8/tapioca-lsp

### CODE REVIEW

File: lib/ruby_lsp/tapioca/addon.rb [hunk 1]
```diff
@@ -62,6 +63,9 @@ def version
 
       sig { params(changes: T::Array[{ uri: String, type: Integer }]).void }
       def workspace_did_change_watched_files(changes)
+        File.write("output.txt", "\nFiles changed", mode: "a")
```
Comment (vinistock) [inline lib/ruby_lsp/tapioca/addon.rb:66]: ✂️

File: lib/ruby_lsp/tapioca/server_addon.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# frozen_string_literal: true
+
+begin
+  # The Tapioca add-on depends on the Rails add-on to add a runtime component to the runtime server. We can allow the
+  # add-on to work outside of a Rails context in the future, but that may require Tapioca spawning its own runtime
+  # server
+  require "ruby-lsp-rails"
+rescue LoadError
+  return
+end
```
Comment (vinistock) [inline lib/ruby_lsp/tapioca/server_addon.rb:11]: Is this necessary given that we don't activate the add-on at all if `ruby-lsp-rails` is not present?

File: lib/ruby_lsp/tapioca/server_addon.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+require "tapioca/internal"
```
Comment (vinistock) [inline lib/ruby_lsp/tapioca/server_addon.rb:13]: Now that the runtime functionality is all here, I think we can remove the internals require from the `addon.rb` file. We're not interacting with Tapioca there, only here.

**Review by vinistock — APPROVED** [review]

File: lib/ruby_lsp/tapioca/addon.rb [hunk 4]
```diff
@@ -62,6 +61,9 @@ def version
 
       sig { params(changes: T::Array[{ uri: String, type: Integer }]).void }
       def workspace_did_change_watched_files(changes)
+        return unless T.must(@global_state).experimental_features
+        return unless @rails_runner_client # Client is not ready
```
Comment (KaanOzkan) [inline lib/ruby_lsp/tapioca/addon.rb:65]: I personally prefer multiple `unless` calls over combining them and thinking of De Morgan's Laws 😬

**Pull request merged**